basyx.aasrepository/basyx.aasrepository-backend/src/main/java/org/eclipse/digitaltwin/basyx/aasrepository/CrudAasRepository.java

```diff
@@ -120,70 +106,75 @@ public void updateAas(String aasId, AssetAdministrationShell aas) {
 
 		throwIfMismatchingIds(aasId, aas);
 
-		aasServices.put(aasId, aasServiceFactory.create(aas));
-	}
-
-	private void throwIfAasExists(AssetAdministrationShell aas) {
-		if (aasServices.containsKey(aas.getId())) {
-			throw new CollidingIdentifierException();
-		}
-	}
-
-	private void throwIfAasDoesNotExist(String aasId) {
-		if (!aasServices.containsKey(aasId)) {
-			throw new ElementDoesNotExistException(aasId);
-		}
+		aasBackend.save(aas);
 	}
 
 	@Override
 	public CursorResult<List<Reference>> getSubmodelReferences(String aasId, PaginationInfo pInfo) {
-		throwIfAasDoesNotExist(aasId);
-
-		CursorResult<List<Reference>> paginatedSubmodelReference = aasServices.get(aasId).getSubmodelReferences(pInfo);
-
-		return paginatedSubmodelReference;
+		return getAasServiceOrThrow(aasId).getSubmodelReferences(pInfo);
 	}
 
 	@Override
 	public void addSubmodelReference(String aasId, Reference submodelReference) {
-		throwIfAasDoesNotExist(aasId);
-		aasServices.get(aasId)
-				.addSubmodelReference(submodelReference);
+		AasService aasService = getAasServiceOrThrow(aasId);
+
+		aasService.addSubmodelReference(submodelReference);
+
+		updateAas(aasId, aasService.getAAS());
 	}
 
 	@Override
 	public void removeSubmodelReference(String aasId, String submodelId) {
-		throwIfAasDoesNotExist(aasId);
-		aasServices.get(aasId)
-				.removeSubmodelReference(submodelId);
+		AasService aasService = getAasServiceOrThrow(aasId);
+
+		aasService.removeSubmodelReference(submodelId);
+
+		updateAas(aasId, aasService.getAAS());
 	}
 
 	@Override
 	public void setAssetInformation(String aasId, AssetInformation aasInfo) throws ElementDoesNotExistException {
-		throwIfAasDoesNotExist(aasId);
-		aasServices.get(aasId)
-				.getAAS()
-				.setAssetInformation(aasInfo);
+		AasService aasService = getAasServiceOrThrow(aasId);
+
+		aasService.setAssetInformation(aasInfo);
+
+		updateAas(aasId, aasService.getAAS());
 	}
 
 	@Override
 	public AssetInformation getAssetInformation(String aasId) throws ElementDoesNotExistException {
-		throwIfAasDoesNotExist(aasId);
-		return aasServices.get(aasId)
-				.getAAS()
-				.getAssetInformation();
+		return getAasServiceOrThrow(aasId).getAssetInformation();
 	}
-  
-	@Override
-	public String getName() {
-		return aasRepositoryName == null ? AasRepository.super.getName() : aasRepositoryName;
+
+	private AasService getAasServiceOrThrow(String aasId) {
+		AssetAdministrationShell aas = aasBackend.findById(aasId).orElseThrow(() -> new ElementDoesNotExistException(aasId));
+
+		AasService aasService = aasServiceFactory.create(aas);
+		return aasService;
```

Comment:
Please use inline expression for return:

return return aasServiceFactory.create(aas);